### PR TITLE
Prepare for release 1.4.0

### PR DESCRIPTION
**Pull request by and-rewsmith**
Updating edgelet refs for release/1.4. Referenced Arnav's PR:
https://github.com/Azure/iotedge/pull/6460

Also contains changes from this PR, which we needed to combine PRs to avoid long checkin times for a time sensitive release:
https://github.com/Azure/iotedge/pull/6623

### CODE REVIEW

**Review by damonbarry — DISMISSED** [review]

**Comment by yophilav** [comment]
1. Please rename the PR title to be `Prepare for Release 1.4.0` 
2. Please only merge this PR once the #6622, #6623, #6624 are merged.

**Review by damonbarry — APPROVED** [review]

**Pull request merged**
